crates/turbopack-ecmascript/src/analyzer/graph.rs

```diff
@@ -1478,6 +1464,42 @@ impl VisitAstPath for Analyzer<'_> {
         self.data.effects = take(&mut self.effects);
     }
 
+    fn visit_try_stmt<'ast: 'r, 'r>(
+        &mut self,
+        stmt: &'ast TryStmt,
+        ast_path: &mut AstNodePath<AstParentNodeRef<'r>>,
+    ) {
+        ast_path.with(
+            AstParentNodeRef::TryStmt(stmt, TryStmtField::Block),
+            |ast_path| {
+                let was_in_try = self.cur_in_try;
+                self.cur_in_try = true;
+
+                stmt.block.visit_with_path(self, ast_path);
+
+                self.cur_in_try = was_in_try;
+            },
+        );
+
+        if let Some(handler) = &stmt.handler {
+            ast_path.with(
+                AstParentNodeRef::TryStmt(stmt, TryStmtField::Handler),
+                |ast_path| {
+                    handler.visit_with_path(self, ast_path);
+                },
+            )
+        }
+
+        if let Some(finalizer) = &stmt.finalizer {
+            ast_path.with(
+                AstParentNodeRef::TryStmt(stmt, TryStmtField::Finalizer),
+                |ast_path| {
+                    finalizer.visit_with_path(self, ast_path);
+                },
+            )
+        }
+    }
```

Comment:
You might also compute the `in_try` flag from `ast_path`.

Should we handle function declarations/expressions, etcs. in try blocks:

``` js
try {
  // in try here
  function x() {
    // not in try here?
  }
  // in try here
} ...
```

Reply:
Didn't think of that, changing it to derive it from the `ast_path`